.github/workflows/python_tests.yml

```diff
@@ -39,11 +39,12 @@ jobs:
           path: ${{ env.pythonLocation }}
           key: ${{ runner.os }}-pip-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('dev-requirements.txt', 'requirements.txt') }}
 
-      - name: Install Dependencies
+      - name: Upgrade pip
         if: steps.python_cache.outputs.cache-hit != 'true'
-        run: |
-          python -m pip install --upgrade pip
-          pip install .[dev]
+        run: python -m pip install --upgrade pip
+
+      - name: Install packages (incl. updated code)
+        run: pip install .[dev]
```

Comment:
But I don't understand, what's the point of using caches then? Should we perhaps try to install `requirements.txt` and `dev-requirements.txt`, force caching these, and only then install `nucypher`?